Framework/Algorithms/src/Rebin.cpp

```diff
@@ -347,12 +336,12 @@ void Rebin::propagateMasks(API::MatrixWorkspace_const_sptr inputWS,
   // Create a zero vector for the errors because we don't care about them here
   const MantidVec zeroes(weights.size(), 0.0);
   // Create a vector to hold the redistributed weights
-  const MantidVec &XValues_new = outputWS->readX(hist);
+  auto &XValues_new = outputWS->x(hist);
   MantidVec newWeights(XValues_new.size() - 1), zeroes2(XValues_new.size() - 1);
   // Use rebin function to redistribute the weights. Note that distribution flag
   // is set
-  VectorHelper::rebin(masked_bins, weights, zeroes, XValues_new, newWeights,
-                      zeroes2, true);
+  VectorHelper::rebin(masked_bins, weights, zeroes, XValues_new.rawData(),
+                      newWeights, zeroes2, true);
```

Comment:
Am I overlooking something or could you also use your new rebin method here if you put the weights into a histogram as frequencies?